lib/getConfigForFile.mjs

```diff
@@ -25,23 +25,25 @@ export default async function getConfigForFile(
 	const optionsConfig = stylelint._options.config;
 	const cwd = stylelint._options.cwd;
 
-	if (optionsConfig !== undefined) {
-		const cached = stylelint._specifiedConfigCache.get(optionsConfig);
+	if (optionsConfig) {
+		const filePathAsCacheKey = filePath ?? '';
```

Comment:
Yes, it's safe. Such a test isn't always necessary since `getConfigForFile()` calls without `filePath` are already performed at various places. Other higher-level test cases cover. Of course, I'm not against having a case in `getConfigForFile.test.mjs` too.

https://github.com/search?q=repo%3Astylelint%2Fstylelint+%22getConfigForFile%28%22+path%3A*.mjs&type=code

We can see also the coverage passes the line of `cachedForFiles.set()`:
https://app.codecov.io/gh/stylelint/stylelint/commit/3fca68706b744b18df53a2691dbd164508cd5d32/blob/lib/getConfigForFile.mjs

Reply:
If you are sure that we can't have a case where we get the wrong `cached` from `cachedForFiles.get('')` because `filePath` was `undefined` for different configs, then we can move on.
I would need to investigate further to tell if it can actually happen.
It seemed unsafe to me at first glance.